### PR TITLE
[stable/traefik] fix: make dashboard ingress optional; add ability to specify custom c…

### PR DESCRIPTION
**Pull request by isindir**
#### What this PR does / why we need it:

- makes dashboard ingress optional
- adds ability to specify custom certificate secret name
- disables TLS cert creation if upstream cert is used or custom cert name is specified
- bumps version of traefik image to 7.11

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

#### Please review

- @emilevauge
- @dtomcej
- @timoreimann
- @vdemeester
- @nmengin
- @mmatur
- @ldez
- @Juliens
- @jbdoumenjou
- @geraldcroes
- @SantoDE
- @errm

### CODE REVIEW

File: stable/traefik/templates/dashboard-ingress.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.dashboard.enabled }}
+{{- if and .Values.dashboard.enabled .Values.dashboard.ingressEnabled }}
```
Comment (dtomcej) [inline stable/traefik/templates/dashboard-ingress.yaml:1]: Not sure I like this logic.

If you set `dashboard.enabled` to `true`, and `dashboard.ingressEnabled` to false, this still executes.

Also, we want the ingress to be the default option, so having to set two flags to enable the default is not ideal.

File: stable/traefik/values.yaml [hunk 2]
```diff
@@ -101,6 +101,7 @@ ssl:
   defaultIPList: []
     # - 1.2.3.4
   # cipherSuites: []
+  customSecretName: ""
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: This should be optional (commented out by default)

File: stable/traefik/values.yaml [hunk 3]
```diff
@@ -310,6 +311,7 @@ dashboard:
   service: {}
     # annotations:
     #   key: value
+  ingressEnabled: true
```
Comment (dtomcej) [inline stable/traefik/values.yaml:314]: This should also be optional (commented out by default)

**Review by dtomcej — CHANGES_REQUESTED** [review]
Not a huge fan of making a bunch of un-related changes in a single PR, because I have some feedback on some changes that will end up blocking other changes that I agree with....

File: stable/traefik/templates/dashboard-ingress.yaml (hunk 1 above)
Comment (ezelenka) [reply stable/traefik/templates/dashboard-ingress.yaml]: I wanted to make it possible to disable dashboard ingress but leaving ability to access dashboard via port forwarding. In a way it is currently done - it is all or none.

File: stable/traefik/values.yaml (hunk 2 above)
Comment (ezelenka) [reply stable/traefik/values.yaml]: I'll update as suggested

File: stable/traefik/templates/dashboard-ingress.yaml (hunk 1 above)
Comment (isindir) [reply stable/traefik/templates/dashboard-ingress.yaml:1]: I wanted to make it possible to disable dashboard ingress but leaving ability to access dashboard via port forwarding. In a way it is currently done - it is all or none. I can take it out from this PR, but I think it is common to be able to enable ingress for some services only if needed.

File: stable/traefik/values.yaml (hunk 2 above)
Comment (isindir) [reply stable/traefik/values.yaml]: updated

File: stable/traefik/values.yaml (hunk 3 above)
Comment (isindir) [reply stable/traefik/values.yaml:314]: The way chart is done - I think it is difficult to achieve without bigger changes, I thought to do less disruption as possible. Unless false becomes default value of cause.

**Pull request closed without merging**